### PR TITLE
Remove style/testing feature

### PR DESCRIPTION
**Pull request by Manishearth**
We added this because a year ago we had no reliable Gecko CI. This meant that Gecko-only properties needed to be tested *somehow*, and we solved that by making it so that for unit tests we compile all properties, not just the servo ones.

This was useful back then, but I don't think we need this anymore. We have reliable Gecko CI, and all the gecko-only stuff we tested is adequately handled by the properties-database parsing mochitests. It's a bit of annoying cruft that just complicates things; we probably should remove it.

r? @emilio or @SimonSapin

---
This change is [Reviewable](https://reviewable.io/reviews/servo/servo/17984)

### CODE REVIEW

**Comment by Manishearth** [comment]
Gah, there's some ruletree stuff that still needs this feature. Kept it around for testing stuff, but removed the whole complexity wrt conditional compilation of properties.

**Review by emilio — COMMENTED** [review]
That's just to avoid asserting thread stuff right? If so, it may be worth to just mock the thread state module or something...

Losing the testing of `font-feature-values` is kinda unfortunate, since we don't test that right now yet...

**Comment by SimonSapin** [comment]
I didn’t realize this would involve removing tests. Can we move them to `tests/units/stylo` instead? Or have a runtime flag instead of a compilation flag?

File: tests/unit/style/rule_tree/bench.rs [hunk 1]
```diff
@@ -148,6 +152,7 @@ fn bench_expensive_insertion(b: &mut Bencher) {
 #[bench]
 fn bench_insertion_basic_parallel(b: &mut Bencher) {
     let r = RuleTree::new();
+    thread_state::initialize(thread_state::SCRIPT);
```
Comment (emilio) [inline tests/unit/style/rule_tree/bench.rs]: I think this would also be fine with `thread_state::LAYOUT`, and would be preferable.

**Review by emilio — COMMENTED** [review]
Looks good... I agree with @SimonSapin that losing test coverage is not great, but I think it's not a _huge_ deal either...

But perhaps it can be moved to `tests/unit/stylo` as he suggests.

**Comment by Manishearth** [comment]
> I didn’t realize this would involve removing tests. 

This is all stuff well tested on the m-c side. Most of these tests are simple tests added by new contributors back when we didn't have mochitests.

Runtime flag won't work, though.

**Comment by SimonSapin** [comment]
Alright. r=me with the commits squashed or reordered. It a previous push the feature was removed before its usage was removed. Now it’s removed, added back, and removed again.

---

Reviewed 17 of 17 files at r1, 14 of 21 files at r2, 9 of 9 files at r3.
Review status: all files reviewed at latest revision, 1 unresolved discussion.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/servo/servo/17984#-:-Kqtu6RmZZhwO0DbG0ut:bxm1qwm)*

**Comment by Manishearth** [comment]
@bors-servo r=SimonSapin

push didn't happen. gah.

**Comment by SimonSapin** [comment]
@bors-servo r-

There’s still stuff like `cfg!(feature = "testing")` being *added* in the second commit. Maybe easier to squash everything?

**Pull request merged**